### PR TITLE
Refs #9162 - removed compatibility workarounds in develop

### PR DESCRIPTION
**Pull request by lzap**
This is going into **develop** only.

### CODE REVIEW

File: lib/foreman_discovery/engine.rb [hunk 1]
```diff
@@ -119,15 +117,8 @@ class Engine < ::Rails::Engine
...
-      # this condition is here for compatibility reason to work with Foreman 1.4.x
-      # Also need to handle the reverse of the 1.8 method above
-      unless (SETTINGS[:version].to_s.include?('develop') or Gem::Version.new(SETTINGS[:version]) >= Gem::Version.new('1.8'))
-        if Apipie.configuration.api_controllers_matcher.is_a?(Array)
-          Apipie.configuration.api_controllers_matcher << "#{ForemanDiscovery::Engine.root}/app/controllers/api/v2/*.rb"
-        end
-      end
-      if Apipie.configuration.respond_to?(:checksum_path)
-        Apipie.configuration.checksum_path += ['/discovered_hosts/']
+      if Apipie.configuration.api_controllers_matcher.is_a?(Array)
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: and no need for this on develop, as apipie_documented_controllers does it

Comment (lzap) [inline lib/foreman_discovery/engine.rb]: Whole if block or just the if statement?

Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: The whole initializer block.

Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: Oh wait, I think you still need `Apipie.configuration.checksum_path` but not the api_controllers_matcher.

**Comment by lzap** [comment]
@GregSutcliffe there are some concerns above, you have tested this you said on IRC. Can you comment on this?

File: lib/foreman_discovery/engine.rb (hunk 1 above)
Comment (lzap) [inline lib/foreman_discovery/engine.rb]: Done. How can I test this... I have no idea.

Comment (domcleal) [inline lib/foreman_discovery/engine.rb]: `rake apipie:cache` and check the docs (in public/ or /apidoc on the server) include discovery's API.

File: lib/foreman_discovery/engine.rb [hunk 2]
```diff
@@ -119,16 +117,8 @@ class Engine < ::Rails::Engine
...
-      # this condition is here for compatibility reason to work with Foreman 1.4.x
-      # Also need to handle the reverse of the 1.8 method above
-      unless (SETTINGS[:version].to_s.include?('develop') or Gem::Version.new(SETTINGS[:version]) >= Gem::Version.new('1.8'))
-        if Apipie.configuration.api_controllers_matcher.is_a?(Array)
-          Apipie.configuration.api_controllers_matcher << "#{ForemanDiscovery::Engine.root}/app/controllers/api/v2/*.rb"
-        end
-      end
       if Apipie.configuration.respond_to?(:checksum_path)
         Apipie.configuration.checksum_path += ['/discovered_hosts/']
-      end
```
Comment (domcleal) [inline lib/foreman_discovery/engine.rb:131]: missing 'end' to the 'if'

Comment (lzap) [inline lib/foreman_discovery/engine.rb:131]: Come on, details... ;-)

**Comment by lzap** [comment]
I've tested this with develop, it generates just fine.

File: lib/foreman_discovery/engine.rb (hunk 2 above)
Comment (domcleal) [inline lib/foreman_discovery/engine.rb:131]: Yeah, I know, you're more of a "big picture" guy ;-)

**Comment by domcleal** [comment]
:+1: looks correct.

**Comment by lzap** [comment]
Since you guys are on the way to FOSDEM and I want to prepare for rc2 today, I am merging this myself.

**Comment by lzap** [comment]
Ok disregard my last comment, this is for `develop`, no need to rush with this one! Please merge later. Have a good FOSDEM!

**Comment by lzap** [comment]
@orrabin when ready please merge, we need to unblock Jenkins (tests of discovery were suspended)

**Comment by domcleal** [comment]
@lzap 6a89b7a4d79369015882a7969a1c9449fac02a4d fixed tests, this just reverts workarounds

**Comment by lzap** [comment]
Rebased, looks sane here.

**Comment by lzap** [comment]
@ehelms we are merging this patch right now, today's nightly should be fixed now and we are about to release discovery 3.0.0 by end of this week.

**Comment by orrabin** [comment]
merged as 2f9a830b23af562d8057a8b4536a53bdea503262, thanks @lzap

**Pull request closed without merging**